### DISCUSSION TITLE
Being redirected when using `up-on-dismissed`

### DISCUSSION THREAD

**Discussion by GabrielNBDS** [unpoly/unpoly#496] [Q&A] [answered]
When I use `up-on-dismissed` I am redirected to the route that I first started.

[Here's a video](https://youtu.be/um34IFXCLcc). First I'm redirected to `/dashboard` and then to `/me`. 

Do anyone have any idea of what is happening?

**Comment by adam12** [comment]
Hard to follow from the video what the issue is. What happens if you remove the `up-on-dismissed` attribute?

Elements in Unpoly know where they've been loaded from, so I wonder if the `up.reload('body')` is using a source location that is not what you expect. Just guessing because I am not really sure.

**Reply by GabrielNBDS** [reply]
If i remove the `up-on-dismissed` it works fine (of course it doesn't reload the layout). I don't know why but when I add it it calls the starting route, if I remove it it doesn't call. I used a `console.log` on the route and it consoled with the `up-on-dismissed` and after I removed it didn't console.

**Reply by adam12** [reply]
Instead of reloading `body`, can you narrow it down to a more specific selector? I'm not sure if it will help but `body` can be a special case.

**Reply by GabrielNBDS** [reply]
I tried to `[layout-sidebar]` and `[layout-main]` as well. It works as expected but instead of loading the url it currently is it makes a request to the starting route. In the video you can see that the url is right but it calls the starting url. It can be seen in the network tab as well.

**Reply by adam12** [reply]
Perhaps see if you can make a reproduction using our [example Glitch project](https://glitch.com/edit/#!/empty-unpoly-project?path=index.html%3A1%3A0).

**Reply by GabrielNBDS** [reply, marked as answer]
When I tried to reproduce it worked so I'm assuming I messed something in that project specifically. Sorry for opening this before trying to reproduce
